### DISCUSSION TITLE
enable-feature=remote-write-receiver

### DISCUSSION THREAD

**Discussion by skinfrakkiirid** [prometheus/prometheus#13090] [General]
I can't get this working at all. I have tried in both the systemd file and the Prometheus config file:

global:
  scrape_interval: 15s
  scrape_timeout: 10s
  evaluation_interval: 15s
alerting:
  alertmanagers:
  - follow_redirects: true
    enable_http2: true
    scheme: http
    timeout: 10s
    api_version: v2
    static_configs:
    - targets: []
scrape_configs:
- job_name: prometheus
  honor_timestamps: true
  scrape_interval: 15s
  scrape_timeout: 10s
  metrics_path: /metrics
  scheme: http
  follow_redirects: true
  enable_http2: true
  static_configs:
  - targets:
    - localhost:9090
remote_write:
- url: http://localhost:9090/api/v1/write
  remote_timeout: 30s
  follow_redirects: true
  enable_http2: true
  queue_config:
    capacity: 10000
    max_shards: 50
    min_shards: 1
    max_samples_per_send: 2000
    batch_send_deadline: 5s
    min_backoff: 30ms
    max_backoff: 5s
  metadata_config:
    send: true
    send_interval: 1m
    max_samples_per_send: 2000

Still get the same result:

curl -X POST http://localhost:9090/api/v1/write
remote write receiver needs to be enabled with --web.enable-remote-write-receiver

**Comment by beorn7** [comment]
`--web.enable-remote-write-receiver` is a command line flag, not an entry in the Prometheus config file. Whatever starts your Prometheus server, you have to tell it to pass in this flag on the command line.

**Reply by skinfrakkiirid** [reply]
Yeah I figured it out. Now I am getting a bunch of these:

caller=write_handler.go:132 level=error component=web msg="Out of order sample from remote write" err="out of order sample" series

**Reply by iamMrGaurav** [reply]
@skinfrakkiirid how did you enable it let me know.

**Comment by beorn7** [comment]
That means that your Prometheus server receives samples out of order. You should find out why you send samples out of order. If it is legitimate, you can set an `out_of_order_time_window` in your config to allow out-of-order ingestion. See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#tsdb . However, you shouldn't use that to gloss over a misconfiguration that creates out-of-order samples accidentally.

**Reply by skinfrakkiirid** [reply]
I'm pushing metrics from Telegraf via the outputs.http plugin. I haven't made any time adjustments. This is all running on the local server for testing purposes. So what would be causing this?

**Reply by beorn7** [reply]
l don't know. Needs investigation. Maybe try to find out if other Telegraf users know about this issue?

**Reply by skinfrakkiirid** [reply]
Hey one other question. My metrics are coming in from most of my Telegraf inputs. However, I have an input.exec that runs a script and inserts data on an InfluxDB database. I see everything in the database so I know the data is there. Is there. However, those metrics aren't in Prometheus. Is there:

1. A way to get those metrics into Prometheus
2. A way from command line to directly insert the metrics into Prometheus. These will include BASH variables

**Reply by skinfrakkiirid** [reply]
BTW I tried adding this to my config. Get this error:

Checking /etc/prometheus/prometheus.yml
  FAILED: parsing YAML file /etc/prometheus/prometheus.yml: yaml: unmarshal errors:
  line 7: field tsdb not found in type config.plain
